File: 451.根据字符出现频率排序.cpp
/*
 * @lc app=leetcode.cn id=451 lang=cpp
 *
 * [451] 根据字符出现频率排序
 */

// @lc code=start
class Solution
{
public:
    string frequencySort(string s)
    {
        string res = "";
        unordered_map<char, int> m;
        for (auto c : s)
            m[c]++;
        map<int, vector<char>> m_map;

        for (auto i : m)
        {
            m_map[i.second].push_back(i.first);
        }

        for (auto i : m_map)
        {

            for (auto e : i.second)
            {
                string str(i.first, e);
                res = str + res;
            }
        }
        return res;
    }
};
// @lc code=end

File: 451.根据字符出现频率排序.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.TreeMap;

/*
 * @lc app=leetcode.cn id=451 lang=java
 *
 * [451] 根据字符出现频率排序
 */

// @lc code=start
class Solution {

    public String frequencySort(String s) {
        Map<Character, Integer> map = new HashMap<>();
        var c = s.toCharArray();
        for (int i = 0; i < c.length; i++) {
            map.put(c[i], map.getOrDefault(c[i], 0) + 1);
        }
        Map<Integer, ArrayList<Character>> treeMap = new TreeMap<>((o1, o2) -> o2 - o1);



    }
}
// @lc code=end
